From: =?utf-8?q?Florian_M=C3=BCllner?= <fmuellner@gnome.org>
Date: Mon, 13 Apr 2020 16:55:34 +0000
Subject: authPrompt: Fix spinner alignment

We want the spinner to be centered with regard to the entry, but
constraining the height breaks that:

 1. clutter_actor_allocate() is called with the available size
 2. clutter_actor_update_constraints() then adjusts that according
    to the constraints
 3. clutter_actor_adjust_allocation() applies the margin/expand/align
    properties.

The issue there is that 2. reduces the allocation to the desired size,
so there is no more extra space to distribute in 3.

We can fix this by either constraining everything (and rely on the
cancel button's alignment) or limit the constraint to the width. The
latter seems more appropriate, given that the constraint is only used
to center the entry horizontally.

(cherry picked from commit b191e9ef9125b3e292b60a5db338f11a75794cc8)

Bug: https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/2628
Origin: upstream, 3.36.2, commit:18a5c7436251d43d458c40a0259dc6da4d9ea760
---
 js/gdm/authPrompt.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/js/gdm/authPrompt.js b/js/gdm/authPrompt.js
index 3434de0..c26b1bc 100644
--- a/js/gdm/authPrompt.js
+++ b/js/gdm/authPrompt.js
@@ -184,7 +184,7 @@ var AuthPrompt = GObject.registerClass({
         });
         this._defaultButtonWell.add_constraint(new Clutter.BindConstraint({
             source: this.cancelButton,
-            coordinate: Clutter.BindCoordinate.SIZE,
+            coordinate: Clutter.BindCoordinate.WIDTH,
         }));
         this._mainBox.add_child(this._defaultButtonWell);
 
